aurora/src/main/java/src/config/JMXConfig.java

```diff
@@ -0,0 +1,141 @@
+package src.config;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.market.banica.common.util.ApplicationDirectoryUtil;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.stereotype.Service;
+import src.model.ChannelProperty;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@EnableMBeanExport
+@ManagedResource
+@Service
+@RequiredArgsConstructor
+public class JMXConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JMXConfig.class);
+
+    private ChannelManager channels;
+
+    private Map<String, ChannelProperty> channelPropertyMap;
```

Comment:
why do you need this map here, instead directly use channel manager

Reply:
When you back-up the file you cannot get from an ManagedChannel the host and the port there is no getter for this so in order to write them back in to the file when map is changed I need to keep them in their first state so I can get the host and the port from the already created channels.

An alternative will be to re-read the file again add the newest adjustment and re-write the file but it will be much slower.